src/ts/components/seekbar.ts

```diff
@@ -607,30 +612,49 @@ export class SeekBar extends Component<SeekBarConfig> {
 
   protected updateMarkers(): void {
     this.seekBarMarkersContainer.empty();
+
+    const seekBarWidthPx = this.seekBar.width();
+
     for (let marker of this.timelineMarkers) {
+      const markerClasses = ['seekbar-marker'].concat(marker.marker.cssClasses || [])
+        .map(cssClass => this.prefixCss(cssClass));
+
+      const cssProperties: {[propertyName: string]: string} = {
+        'width': marker.position + '%',
+      };
+
+      if (marker.duration > 0) {
+        const markerWidthPx = Math.round(seekBarWidthPx / 100 * marker.duration);
+        cssProperties['border-right-width'] = markerWidthPx + 'px';
+        cssProperties['margin-left'] = '0';
+      }
+
       this.seekBarMarkersContainer.append(new DOM('div', {
-        'class': this.prefixCss('seekbar-marker'),
-        'data-marker-time': String(marker.time),
-        'data-marker-title': String(marker.title),
-      }).css({
-        'width': marker.time + '%',
-      }));
+        'class': markerClasses.join(' '),
+        'data-marker-time': String(marker.marker.time),
+        'data-marker-title': String(marker.marker.title),
+      }).css(cssProperties));
     }
   }
 
-  protected getMarkerAtPosition(percentage: number): TimelineMarker | null {
-    let snappedMarker: TimelineMarker = null;
-    let snappingRange = 1;
+  protected getMarkerAtPosition(percentage: number): SeekBarMarker | null {
```

Comment:
should it be possible to have overlapping markers? if so we must not break in the for loop bur rather collect all matching markers

Reply:
It is possible (overlaps are not prevented), but it does not make sense and I leave the responsibility to manage that correctly to the user. Also, which `SeekBarLabel` for which marker should be displayed then? All of them? Let's also leave this open until it becomes an issue.